federation-2/harmonizer/js-src/composition.ts

```diff
@@ -0,0 +1,38 @@
+import { composeServices, CompositionResult } from "@apollo/composition";
+import { parse } from "graphql";
+
+export function composition(
+  serviceList: { sdl: string; name: string; url?: string }[]
+): CompositionResult {
+  if (!serviceList || !Array.isArray(serviceList)) {
+    throw new Error("Error in JS-Rust-land: serviceList missing or incorrect.");
+  }
+
+  serviceList.some((service) => {
+    if (
+      typeof service.name !== "string" ||
+      !service.name ||
+      (typeof service.url !== "string" && service.url) ||
+      (typeof service.sdl !== "string" && service.sdl)
```

Comment:
we probably don't need these runtime checks now that we have typescript involved. what do you think?